src/Roslyn.Diagnostics.Analyzers/Core/ImportingConstructorShouldBeObsoleteCodeFixProvider.cs

```diff
@@ -0,0 +1,270 @@
+                        createChangedDocument = cancellationToken => AddObsoleteAttributeAsync(context.Document, diagnostic.Location.SourceSpan, cancellationToken);
+                        break;
+
+                    case ImportingConstructorShouldBeObsolete.Scenario.MissingDescription:
+                        title = RoslynDiagnosticsAnalyzersResources.ImportingConstructorShouldBeObsoleteCodeFix_MissingDescription;
+                        createChangedDocument = cancellationToken => AddDescriptionAndErrorAsync(context.Document, diagnostic.Location.SourceSpan, cancellationToken);
+                        break;
+
+                    case ImportingConstructorShouldBeObsolete.Scenario.IncorrectDescription:
+                        title = RoslynDiagnosticsAnalyzersResources.ImportingConstructorShouldBeObsoleteCodeFix_IncorrectDescription;
+                        createChangedDocument = cancellationToken => UpdateDescriptionAsync(context.Document, diagnostic.Location.SourceSpan, cancellationToken);
+                        break;
+
+                    case ImportingConstructorShouldBeObsolete.Scenario.MissingError:
+                        title = RoslynDiagnosticsAnalyzersResources.ImportingConstructorShouldBeObsoleteCodeFix_MissingError;
+                        createChangedDocument = cancellationToken => AddErrorAsync(context.Document, diagnostic.Location.SourceSpan, cancellationToken);
+                        break;
+
+                    case ImportingConstructorShouldBeObsolete.Scenario.ErrorSetToFalse:
+                        title = RoslynDiagnosticsAnalyzersResources.ImportingConstructorShouldBeObsoleteCodeFix_ErrorSetToFalse;
+                        createChangedDocument = cancellationToken => SetErrorToTrueAsync(context.Document, diagnostic.Location.SourceSpan, cancellationToken);
+                        break;
+
+                    default:
+                        continue;
+                }
+
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        title,
+                        createChangedDocument,
+                        equivalenceKey: scenario),
+                    diagnostic);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private async Task<Document> AddObsoleteAttributeAsync(Document document, TextSpan sourceSpan, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var obsoleteAttributeSymbol = semanticModel.Compilation.GetTypeByMetadataName(typeof(ObsoleteAttribute).FullName);
+            if (obsoleteAttributeSymbol is null)
+            {
+                return document;
+            }
+
+            var constructor = root.FindNode(sourceSpan, getInnermostNodeForTie: true);
```

Comment:
Handle null `constructor`?

Reply:
Not possible unless the code fix infrastructure is broken; leaving as is since the invalid state exception is already handled by the infrastructure